packages/commerce-sdk/src/core/loyalty/index.ts

```diff
@@ -0,0 +1,213 @@
+import { SDKOptions, ThirdwebSDK } from "@thirdweb-dev/sdk";
+import crypto from "crypto";
+import { NextApiRequest } from "next";
+import { LoyaltyRewardsParams, ResponseBody, RewardTokensParams } from "../../../types";
+import { GET_ORDER_BY_ID_QUERY } from "../../lib/queries";
+import { getRawBody, shopifyFetchAdminAPI } from '../../lib/utils';
+
+/**
+ * Gift ERC20 loyalty tokens to a customer using their wallet address.
+ * 
+ * @description This function is used to gift loyalty tokens to a customer using their wallet address. The customer must have purchased and item from your store in order to receive the loyalty tokens. The function will check the order ID and verify that the customer has purchased an item from your store. If the customer has purchased an item from your store, the function will gift the customer the loyalty tokens.
+ * 
+ * @example
+ * ```
+ * const txHash = await rewardPoints({
+    nextApiRequest: {{next_api_request}},
+    rewardAmount: 100,
+    tokenContractAddress: {{"contract_address"}},
+    webhookSecret: {{"webhook_secret"}},
+    chain: "goerli",
+    privateKey: {{"private_key"}},
+    shopifyAdminUrl: {{"shopify_admin_url"}},
+    shopifyAccessToken: {{"shopify_access_token"}},
+  });
+ * ```
+ * @returns Transaction hash
+ * @public
+ */
+
+export async function rewardTokensOnPurchase({
+  nextApiRequest,
+  webhookSecret,
+  tokenContractAddress,
+  gaslessRelayerUrl,
+  chain,
+  rewardAmount,
+  privateKey,
+  shopifyAdminUrl,
+  shopifyAccessToken,
+}: LoyaltyRewardsParams) {
+  const secretKey = webhookSecret;
+
+  const hmac = nextApiRequest.headers["x-shopify-hmac-sha256"];
+  const body = await getRawBody(nextApiRequest);
+  const hash = crypto
+    .createHmac("sha256", secretKey)
+    .update(body)
+    .digest("base64");
+
+    // Compare our hash to Shopify's hash
+    if (hash === hmac) {
+      const shopifyOrderId = nextApiRequest.headers["x-shopify-order-id"];
```

Comment:
looks like all we need from the inboud request are the order ID + the hmac -> can we just let users pass us those? we could then still build helpers for next api routes / express / etc to grab them off of the request for the user, but I'd like to split that out